### ISSUE TITLE
CTD in First Few Turns

### ISSUE THREAD

**Issue by Glidergun** [LoneGazebo/Community-Patch-DLL#1964]
Bug Report Template

Mod Version (i.e Date - (4/23b)):

6-14

1. Mod List (if using standard CPP set, leave blank):

CP and CBO only

2. Type of error (i.e. crash, interface bug, AI quirk): 

Crash to desktop

3. Steps to reproduce:

Sometime on turn 2 or 3, after hitting the End Turn button, the game will crash during an AI turn. Happens, as far as I can tell, regardless of the primary set-up choices (i.e. map, type, size, gamespeed). However, unlike the other reported version of this crash, did not happen when taking ancient ruins.

4. Additional information:

VP was installed with the .exe autoinstaller, I have deleted cache, verified the base game's integrity, reinstalles VP multiple times, basically everything of that nature except completely reinstalling Civ V.

I have all DLC.

---------------------------
Supporting information:

Please note that you can attach .zip files to Issues by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files: Database.log and Lua.log needed
2. Minidump file (located in your Civ5.exe directory)
3. Screenshots (if needed)

[CP and CBO crash files.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/316299/CP.and.CBO.crash.files.zip)

**Comment by Mcantona** [comment]
Practically the same happens to me:

Bug Report Template

Mod Version (i.e Date - (4/23b)):

6-14

Mod List (if using standard CPP set, leave blank):


Type of error (i.e. crash, interface bug, AI quirk):
Crash to desktop after 2nd turn (during AI turns)

Steps to reproduce:
Tried several times with different civs. Always happens after i finish my second turn. I had no problems with the previous version (03-)

Additional information:
VP was installed with the .exe autoinstaller, I have deleted cache, verified the base game's integrity, reinstalles VP multiple times.

I have all DLC.

[ctd_files.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/316572/ctd_files.zip)

**Comment by cononderbee** [comment]
Having the same issues, taken the same precautions to eliminate other issues.


https://drive.google.com/file/d/0B2Q1et4fsZqMVVhDbkpCck5oRGc/view?usp=sharing

**Comment by Shadow-killer56** [comment]
I am also having the same problem with the consistent crashes.

Research tree is also messed up a bit.
[Civ5 Logs and Dmp.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/316881/Civ5.Logs.and.Dmp.zip)
![civ5 research tree bug](https://cloud.githubusercontent.com/assets/19958498/16092395/9c23f258-332f-11e6-9317-0d78296aeb2e.png)

**Comment by LoneGazebo** [comment]
https://github.com/LoneGazebo/Community-Patch-DLL/issues/1962

**Comment by LoneGazebo** [comment]
@ilteroi, all these CTDs point to that same spot I noted in the HomelandAI - the patrol moves iterator seems bad. Dunno why.

**Comment by LoneGazebo** [comment]
Regarding tech screen, I have no idea. Does not do that on my end, and your lua logs are inconclusive.

**Comment by Jotaro31400** [comment]
Mod version : 

06.14

Mod list : 

VP full install + No Ice on coast mod

Type of error : 

Crash on desktop

Lua + database log
[Logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/317112/Logs.zip)

MiniDump
[CvMiniDump.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/317115/CvMiniDump.zip)

Steps to reproduce : 

Any civs, I ctd after 2 or 3 turns after the start, during AI turns

**Comment by Scotfree2** [comment]
Mod version:

06.14

Mod list:

Just CP turned on for this test

Type of error:

Crash to Desktop

Logs + Dump file: 
[Logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/317266/Logs.zip)

Steps to reproduce:
Play the game, crash on an early turn

--------
I tried a couple mod combinations (all CP 1-6a mods, then just CP 1 mod), crashed in the first 2-3 turns every time. I then tried a non-modded game to make sure it would run, it ran for about 20 turns, then returned to main menu, attempted to start a CP 1 mod only, crashed on game setup this time. Attempted one more time (this log set), and made it to the 9th turn before crashing.

**Comment by ilteroi** [comment]
cant reproduce ...

but i captured a settler from denmark and got the return unit popup. that doesn't make sense.

**Comment by ilteroi** [comment]
@G: the patrol moves had been disabled for quite a while, i re-enabled them recently. don't have any problems with them, but they are not essential, so if you need to fix this quickly, just skip them.

**Comment by LoneGazebo** [comment]
@ilteroi was the settler owned by someone else beforehand, or was it to return to him? Odd, that code hasn't been touched in forever...

Re: patrols, I'm not sure, though. The code looks identical to the other functions - what would cause that one, specifically, to break? Are we sure it isn't just because it is coming up first?

**Comment by ilteroi** [comment]
no the settler was an original capture

patrol moves are dead last in the priority list

**Comment by cdanger341** [comment]
I see my original report was merged into this one. I just wanted to add that I played around some more and determined that RED Modpack appears to not be the culprit. CTD still occurs with RED removed and a fresh installation of Vox. HOWEVER, I noticed something new that seems to be related...

I started as a Civ with no special bonuses to their Warrior units and started scouting the area around my first city, like you do. I noticed that after moving the Warrior unit they were tagged as having been given a multi-turn move order (as if I had clicked for them to move further than their move allowance). I clicked the "Cancel Last Mission" button to clear it, and BOOM, no CTD on Turn 1. I tried this again each turn (As every time I moved the Warrior unit, even though it was only 1 or 2 hexes, they were again tagged as having a multi-turn move order), and each time manually de-toggled the "Move Order Mission". I was able to proceed with playing for several more turns until I discovered a Ruin and moved onto it. This time I got the CTD.

I also checked my Tech Tree, but it appeared to be functioning as normal. I did not have the scrolling issue some others have reported. So there's definitely something screwy going on with the movement code/AI.

Where can I find these .lua files, and would it help to include them at this point?
